### PR TITLE
[pack] throwing exception if no version property in host.json does not equal '2.0'

### PR DESCRIPTION
**Pull request by brettsam**

### CODE REVIEW

File: src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs [hunk 1]
```diff
@@ -195,6 +211,15 @@ private IHost BuildHost(bool isOffline = false)
                 });
             }
 
+            if (hasHostConfigErrors)
+            {
+                builder.ConfigureAppConfiguration(b =>
+                {
+                    var source = b.Sources.OfType<HostJsonFileConfigurationSource>().Single();
```
Comment (fabiocav) [inline src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs]: Can we flow a configuration item that could be used by the provider instead of removing it from the collection?

File: src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs [hunk 2]
```diff
@@ -83,16 +84,31 @@ private async Task StartHostAsync(CancellationToken cancellationToken, int attem
             try
             {
                 bool isOffline = CheckAppOffline();
+                bool hasHostConfigError = false;
 
-                _host = BuildHost(isOffline);
+                try
```
Comment (fabiocav) [inline src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs]: Can we avoid the nesting here? It would be good to avoid duplicating the code that sets the exceptions and other flags. 

Perhaps check for that exception type in our existing catch block and conditionally take the recovery action if the exception type is `HostConfigurationException`?

File: src/WebJobs.Script.WebHost/WebHostServiceCollectionExtensions.cs [hunk 3]
```diff
@@ -6,6 +6,7 @@
 using Microsoft.AspNetCore.Authorization;
 using Microsoft.Azure.WebJobs.Extensions.Http;
 using Microsoft.Azure.WebJobs.Script.Config;
+using Microsoft.Azure.WebJobs.Script.Eventing;
```
Comment (mathewc) [inline src/WebJobs.Script.WebHost/WebHostServiceCollectionExtensions.cs]: ?

File: src/WebJobs.Script.WebHost/WebJobsApplicationBuilderExtension.cs [hunk 4]
```diff
@@ -46,7 +47,10 @@ public static IApplicationBuilder UseWebJobsScriptHost(this IApplicationBuilder
             builder.UseMiddleware<HttpExceptionMiddleware>();
             builder.UseMiddleware<ResponseBufferingMiddleware>();
             builder.UseMiddleware<HomepageMiddleware>();
-            builder.UseMiddleware<HttpThrottleMiddleware>();
+            builder.UseWhen(context => context.Features.Get<IFunctionExecutionFeature>() != null, config =>
+            {
```
Comment (mathewc) [inline src/WebJobs.Script.WebHost/WebJobsApplicationBuilderExtension.cs]: Why this change?

Comment (brettsam) [reply src/WebJobs.Script.WebHost/WebJobsApplicationBuilderExtension.cs]: If the JobHost wasn't running (or was waiting due to an error and was in its backoff retry), we were unable to hit admin routes b/c this Middleware needed the HttpRequestQueue, which only comes from the JobHost. @fabiocav and I thought that this was an okay approach since we hadn't want to throttled on admin routes previously anyway.

Comment (fabiocav) [reply src/WebJobs.Script.WebHost/WebJobsApplicationBuilderExtension.cs]: Yeah, this gets us back to the previous behavior, where only functions are throttled
